service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/CaseSubmissionControllerSubmittedTest.java

```diff
@@ -273,39 +270,42 @@ private CaseDetails enableSendToCtscOnCaseDetails(YesNo enableCtsc) {
             ))).build();
     }
 
-    private ImmutableMap.Builder<String, Object> getCompleteParameters() {
+    private <T extends SharedNotifyTemplate> T getCompleteParameters(T template) {
         List<String> ordersAndDirections = List.of("Emergency protection order", "Contact with any named person");
-
-        return ImmutableMap.<String, Object>builder()
-            .putAll(getCommonParameters())
-            .put("dataPresent", "Yes")
-            .put("fullStop", "No")
-            .put("ordersAndDirections", ordersAndDirections)
-            .put("timeFramePresent", "Yes")
-            .put("timeFrameValue", "same day")
-            .put("urgentHearing", "Yes")
-            .put("nonUrgentHearing", "No")
-            .put("firstRespondentName", "Smith");
+        T updatedTemplate = setCommonTemplateParameters(template);
+
+        updatedTemplate.setDataPresent(YES.getValue());
+        updatedTemplate.setFullStop(NO.getValue());
+        updatedTemplate.setOrdersAndDirections(ordersAndDirections);
+        updatedTemplate.setTimeFramePresent(YES.getValue());
+        updatedTemplate.setTimeFrameValue("same day");
+        updatedTemplate.setUrgentHearing(YES.getValue());
+        updatedTemplate.setNonUrgentHearing(NO.getValue());
+        updatedTemplate.setFirstRespondentName("Smith");
+
+        return updatedTemplate;
     }
 
-    private ImmutableMap.Builder<String, Object> getInCompleteParameters() {
-        return ImmutableMap.<String, Object>builder()
-            .putAll(getCommonParameters())
-            .put("dataPresent", "No")
-            .put("fullStop", "Yes")
-            .put("ordersAndDirections", "")
-            .put("timeFramePresent", "No")
-            .put("timeFrameValue", "")
-            .put("urgentHearing", "No")
-            .put("nonUrgentHearing", "No")
-            .put("firstRespondentName", "");
+    private <T extends SharedNotifyTemplate> T getInCompleteParameters(T template) {
+        T updatedTemplate = setCommonTemplateParameters(template);
+
+        updatedTemplate.setDataPresent(NO.getValue());
+        updatedTemplate.setFullStop(YES.getValue());
+        updatedTemplate.setOrdersAndDirections(List.of(""));
+        updatedTemplate.setTimeFramePresent(NO.getValue());
+        updatedTemplate.setTimeFrameValue("");
+        updatedTemplate.setUrgentHearing(NO.getValue());
+        updatedTemplate.setNonUrgentHearing(NO.getValue());
+        updatedTemplate.setFirstRespondentName("");
+
+        return updatedTemplate;
     }
 
-    private Map<String, Object> getCommonParameters() {
-        return Map.of(
-            "localAuthority", "Example Local Authority",
-            "reference", CASE_REFERENCE.toString(),
-            "caseUrl", String.format("http://fake-url/case/%s/%s/%s", JURISDICTION, CASE_TYPE, CASE_REFERENCE)
-        );
+    private <T extends SharedNotifyTemplate> T setCommonTemplateParameters(T template) {
```

Comment:
You don't need to return anything for this method if you want you are going to be effecting the same object so unless you call this method directly from the test you don't need to return the object as the prior method will already have the reference for it